src/System.Linq/src/System/Linq/Intersect.cs

```diff
@@ -19,8 +19,14 @@ public static IEnumerable<TSource> Intersect<TSource>(this IEnumerable<TSource>
             {
                 throw Error.ArgumentNull(nameof(second));
             }
-
-            return IntersectIterator(first, second, null);
+           
+            foreach (TSource element in first)
```

Comment:
There's no guard against `first` containing multiple values that are in `second` and returning them each time instead of just once.
(If this passed tests, we have a gap in the testing).